plugins/authentication/before-request.js

```diff
@@ -3,6 +3,10 @@ module.exports = authenticationBeforeRequest
 const btoa = require('btoa-lite')
 
 function authenticationBeforeRequest (state, options) {
+  if (state.otp) {
+    options.headers['x-github-otp'] = state.otp
+  }
+
```

Comment:
maybe check for `state.otp` below after we check for `state.auth.username`, as it’s only relevant in combination with Basic authentication?

Reply:
Makes sense! Moved it below: https://github.com/octokit/rest.js/pull/1281/commits/856a7912786a75f7a9cf1b440cfd43ea5ac6b41e.